x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -231,3 +184,51 @@ export async function getDocumentSources({
     hasDurationSummaryField: false,
   });
 }
+
+async function getDurationSummarySupportByDocType({
+  apmEventClient,
+  documentTypes,
+  filters,
+  enableContinuousRollups,
+}: {
+  apmEventClient: APMEventClient;
+  documentTypes: ApmDocumentType[];
+  filters: estypes.QueryDslQueryContainer[];
+  enableContinuousRollups: boolean;
+}) {
+  const sources = documentTypes.flatMap((documentType) => {
+    const docTypeConfig = getConfigForDocumentType(documentType);
+
+    return (
+      enableContinuousRollups
+        ? docTypeConfig.rollupIntervals
+        : [RollupInterval.OneMinute]
+    ).flatMap((rollupInterval) => ({
```

Comment:
**doubt:** `flatMap` should expect an array as a return value for its callback, but here an object is returned instead. How is this behaving with the returned object?

Reply:
Thanks for the suggestion. this whole rollover function can be removed, though. I forgot to do it before reopening the PR.

Reply:
The inner `flatMap` is wrong. Thanks for catching it.